### ISSUE TITLE
Integrating MarkItUp for text field inside a BlockService triggers unwanted Drag & Drop

### ISSUE THREAD

**Issue by vlaurier** [sonata-project/SonataPageBundle#851]
### Environment
Debian GNU/Linux 8 (jessie)

#### Sonata packages

sonata-project/page-bundle         3.4.1
sonata-project/formatter-bundle   3.2.1

#### Symfony packages

symfony/symfony 2.8.21

#### PHP version

PHP 5.6.30-0+deb8u1

## Subject

When I try to integrate MarkItUp (used in FormatterBundle) for a textarea **inside a block**, the following occurs:

Clicking on one of the MarkItUp button and moving the mouse...
![before-clicking-onh3](https://user-images.githubusercontent.com/3454840/29357453-ea74cec6-8277-11e7-936f-536c5698f4ae.png)

...triggers a drag and drop event.
![after-clicking-and-moving](https://user-images.githubusercontent.com/3454840/29357470-f73d129e-8277-11e7-9716-3eac1c586592.png)

## Steps to reproduce

- Integrate MarkItDown using the FormatterBundle
- Use the SimpleFormatterType inside a BlockService

Code sample:

```php
    public function buildEditForm(FormMapper $formMapper, BlockInterface $block)
    {
        $formMapper->add('settings', 'sonata_type_immutable_array', [
            'keys' => [
                ['content', SimpleFormatterType::class, [
                    'format' => 'markdown',
                ]],
            ],
        ]);
    }
```
## Expected results

No Drag and Drop interaction should be triggered.

## Actual results

See images above.

**Comment by xXNoceboXx** [comment]
# **Confirmation of issue**

I have the same issue with others versions of Page bundle and Formatter bundle.

## **Images**

![issue_rte_1](https://user-images.githubusercontent.com/3227304/29364396-8ab28174-8293-11e7-936c-68cc81f39fe2.png)

After click one button or extend size of textarea in RTE : 

![issue_rte_3](https://user-images.githubusercontent.com/3227304/29364466-cc69fa52-8293-11e7-8ad1-513ed33e02ce.png)

# **Environment**

Ubuntu 16.04 LTS

## **Sonata packages**

sonata-project/page-bundle 3.5.1
sonata-project/formatter-bundle 3.2.2

## **Symfony packages**

symfony/symfony 2.8.26

**## PHP version**

PHP 5.6.30

**Comment by vlaurier** [comment]
I would like to share the fix I found, because it could be usefull to others.
There is a conflict between MarkItUp library and composer.js from PageBundle. Not sure if the fix should/could be integrated in one of those two librairies. I ended up with adding a custom js code inside a $(document).ready function:

        document.querySelector('body').addEventListener('DOMNodeInserted', function(event) {
            if ($(event.target).hasClass('markItUpHeader')) {
                $(event.target).mousedown(function(){ return false; });
            }
        });

The code prevents Drag and Drop to occur when MarkItUp buttons are clicked.
